browser/src/layer/marker/TextInput.js

```diff
@@ -289,16 +290,32 @@ L.TextInput = L.Layer.extend({
 	// Marks the content of the textarea/contenteditable as selected,
 	// for system clipboard interaction.
 	select: function() {
-		this._textArea.select();
+		this._setSelectionRange(0, this.getPlainTextContent().length);
 	},
 
 	getValue: function() {
-		var value = this._textArea.value;
+		var value = this.getPlainTextContent();
 		if (this._map.formulabar && this._map.formulabar.hasFocus())
-			value = this._map.formulabar.getValue();
+			value =  ' ' + this._map.formulabar.getValue() + ' ';
```

Comment:
Curious about the extra spaces here really =)

Reply:
If you look at changes in `Control.FormulaBarJSDialog.js` you can see that extra spaces were already there. I just moved the code in TextInput since refactoring is more difficult when encapsulation is broken. As far as the reason extra spaces are needed is that `_onInput` implementation expects them. More on next comment.

Reply:
Extra spaces removed in new commit version